src/two_player_collision_example.cpp

```diff
@@ -0,0 +1,357 @@
+      kNominalHeadingCostWeight, kP1HeadingIdx, kP1NominalHeading,
+      "NominalHeadingP1");
+  // p1_cost.AddStateCost(p1_nominal_orientation_cost);
+  const auto p2_nominal_orientation_cost = std::make_shared<OrientationCost>(
+      kNominalHeadingCostWeight, kP2HeadingIdx, kP1NominalHeading,
+      "NominalHeadingP2");
+  // p2_cost.AddStateCost(p2_nominal_orientation_cost);
+  
+  // cost for deviating from the center of the lane (for both p1 and p2)
+  const Polyline2 lane1_p1p2(
+      {Point2(2.5, -50.0), Point2(2.5, 50.0)});
+  const std::shared_ptr<QuadraticPolyline2Cost> p1_lane1_cost(
+      new QuadraticPolyline2Cost(kLaneCostWeight, lane1_p1p2, {kP1XIdx, kP1YIdx},
+                                 "LaneCenter"));
+  const std::shared_ptr<QuadraticPolyline2Cost> p2_lane1_cost(
+      new QuadraticPolyline2Cost(kLaneCostWeight * 10, lane1_p1p2, {kP2XIdx, kP2YIdx},
+                                 "LaneCenter"));
+  p1_cost.AddStateCost(p1_lane1_cost);
+  p2_cost.AddStateCost(p2_lane1_cost);
+
+  // cost for leaving the left boundary of the lane (for both p1 and p2)
+  const std::shared_ptr<SemiquadraticPolyline2Cost> p1_lane1_l_cost(
+      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight * 1000, lane1_p1p2,
+                                     {kP1XIdx, kP1YIdx}, -kLaneHalfWidth,
+                                     !kOrientedRight, "LaneLeftBoundary"));
+  const std::shared_ptr<SemiquadraticPolyline2Cost> p2_lane1_l_cost(
+      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight * 10, lane1_p1p2,
+                                     {kP2XIdx, kP2YIdx}, -kLaneHalfWidth,
+                                     !kOrientedRight, "LaneLeftBoundary"));
+  p1_cost.AddStateCost(p1_lane1_l_cost);
+  p2_cost.AddStateCost(p2_lane1_l_cost);
+
+  // p2 cost of leaving right lane boundary
+  const std::shared_ptr<SemiquadraticPolyline2Cost> p2_lane1_r_cost(
+      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight, lane1_p1p2,
+                                     {kP2XIdx, kP2YIdx}, kLaneHalfWidth,
+                                     kOrientedRight, "LaneRightBoundary"));
+  p2_cost.AddStateCost(p2_lane1_r_cost);
+
+  // p1 right lane boundary cost
+  const Polyline2 lane1_p1(
+      {Point2(2.5 + kLaneHalfWidth, -50.0), Point2(2.5 + kLaneHalfWidth, -5.0)});
+  const Polyline2 lane2_p1(
+      {Point2(2.5 + kLaneHalfWidth, 5.0), Point2(2.5 + kLaneHalfWidth, 50.0)});
+  const Polyline2 lane3_p1(
+      {Point2(10.0, -5.0), Point2(10.0, 5.0)});
+  const Polyline2 lane4_p1(
+      {Point2(2.5 + kLaneHalfWidth, 5.0), Point2(25.0, 5.0)});
+  const Polyline2 lane5_p1(
+      {Point2(2.5 + kLaneHalfWidth, -5.0), Point2(25, -5.0)});
+  const std::shared_ptr<SemiquadraticPolyline2Cost> p1_lane2_r_cost(
+      new SemiquadraticPolyline2Cost(kLaneBoundaryCostWeight, lane1_p1,
+                                     {kP1XIdx, kP1YIdx}, 0.0,
+                                     kOrientedRight, "LaneRightBoundary_lane1_p1"));
```

Comment:
You don't have to include p1 in the cost name, since it's already only being added to `p1_cost` and will show up on the GUI under the dropdown menu for p1.